pkg/backend/httpstate/backend.go

```diff
@@ -828,13 +828,13 @@ func (b *cloudBackend) runEngineAction(
 	<-displayDone
```

Comment:
shouldn't pulling on displayDone be done after pulling on eventsDone... ?

Reply:
talking this over, i think this code is confusing and could be improved. :)  but it's also likely not that important to change.  so we'll go with the targetted fix to address the race condition.